evcouplings/visualize/pairs.py

```diff
@@ -339,6 +353,146 @@ def plot_contact_map(ecs=None, monomer=None, multimer=None,
+            min(intra2_boundaries[0][0],inter_boundaries[1][0]),
+            max(intra2_boundaries[0][1],inter_boundaries[1][1])
+        )
+    ]
+
+    # Calculate the length ratios of the monomers
+    mon1_len = intra1_boundaries[0][1] - intra1_boundaries[0][0]
+    mon2_len = intra2_boundaries[0][1] - intra2_boundaries[0][0]
+
+    if (mon1_len == 0) and (mon2_len ==0):
+        raise ValueError("Warning, you must provide at least one contact to plot "+
+            "for at least one of the monomers. Contact map not generated.")
+
+    ratio1 = mon1_len / (mon1_len + mon2_len)
+    ratio2 = mon2_len / (mon1_len + mon2_len)
+
+    # Initiate the axes using the above ratios
+    fig = plt.figure(figsize=(8, 8))
+    gs = gridspec.GridSpec(2, 2,
+                           width_ratios=[ratio1, ratio2],
+                           height_ratios=[ratio1, ratio2]
+                           )
+    ax1 = plt.subplot(gs[0])  # intra 1, upper left
+    ax2 = plt.subplot(gs[1])  # inter, upper right
+    ax3 = plt.subplot(gs[2])  # inter, lower left
+    ax4 = plt.subplot(gs[3])  # intra 2, lower right
+
+    # intra 1, upper left
+    if not (intra1_ecs is None and d_intra_i is None and d_multimer_i is None):
+        new_kwargs = deepcopy(kwargs)
+        new_kwargs["boundaries"] = intra1_boundaries
+        plot_contact_map(ax=ax1,
+                         symmetric=True,
+                         ecs=intra1_ecs,
+                         monomer=d_intra_i,
+                         multimer=d_multimer_i,
+                         **new_kwargs)
+
+    # intra 2, lower right
+    if not (intra2_ecs is None and d_intra_j is None and d_multimer_j is None):
+        new_kwargs = deepcopy(kwargs)
+        new_kwargs["boundaries"] = intra2_boundaries
+        plot_contact_map(ax=ax4,
+                         symmetric=True,
+                         ecs=intra2_ecs,
+                         monomer=d_intra_j,
+                         multimer=d_multimer_j,
+                         **new_kwargs)
+
+    # inter, lower left
+    if not (inter_ecs is None and d_inter is None):
+        new_kwargs = deepcopy(kwargs)
+        new_kwargs["boundaries"] = inter_boundaries
+        plot_contact_map(ax=ax3,
+                         symmetric=False,
+                         ecs=inter_ecs,
+                         multimer=d_inter,
+                         distance_cutoff=8,
+                         **new_kwargs)
+
+    # inter, upper right
+    if inter_ecs is None:
+        inter_ecs_transposed = None
+    else:
+        inter_ecs_transposed = inter_ecs.rename(columns={"i":"j","j":"i"})
+
+    if d_inter is None:
+        d_inter_T = None
+    else:
+        d_inter_T = d_inter.transpose()
+
+    new_kwargs = deepcopy(kwargs)
+    new_kwargs["boundaries"] = list(reversed(inter_boundaries))
+
+    plot_contact_map(ax=ax2,
+                     symmetric=False,
+                     ecs=inter_ecs_transposed,
+                     multimer=d_inter_T,
```

Comment:
Yeah I did it for the coloring, and I guess because I was thinking that inter molecular contacts were more akin to multimer contacts than to monomer contacts so it made sense to inherit the styles. I can see the argument for doing it the other way though.